sky/data/skystorage.py

```diff
@@ -0,0 +1,334 @@
+"""Skystorage module"""
+import os
+import pathlib
+import subprocess
+import time
+from typing import Any, List, Optional
+import urllib.parse
+
+import click
+import psutil
+
+from sky import sky_logging
+from sky.utils import common_utils
+from sky.utils import db_utils
+
+logger = sky_logging.init_logger(__name__)
+
+_CSYNC_BASE_PATH = '~/.skystorage'
+
+_DB_PATH = os.path.expanduser('~/.sky/skystorage.db')
+pathlib.Path(_DB_PATH).parents[0].mkdir(parents=True, exist_ok=True)
+
+
+def create_table(cursor, conn):
+    cursor.execute("""\
+        CREATE TABLE IF NOT EXISTS running_csync (
+        csync_pid INTEGER PRIMARY KEY,
+        sync_pid INTEGER DEFAULT 0,
+        source_path TEXT)""")
+
+    conn.commit()
+
+
+_DB = db_utils.SQLiteConn(_DB_PATH, create_table)
+_CURSOR = _DB.cursor
+_CONN = _DB.conn
```

Comment:
We probably don't want to create this table any time this file is imported. One way is to add an indirect function, and have all the following functions that uses the database wrapped by the decorator `@db`:
```suggestion
_DB = None
_CURSOR = None
_CONN = None

def db(func):
    @functools.wrap(func)
    def wrapper(*args, **kwargs):
        if _DB is not None:
            return
        
        _DB_PATH = os.path.expanduser('~/.sky/skystorage.db')
        pathlib.Path(_DB_PATH).parents[0].mkdir(parents=True, exist_ok=True)
        
        
        def create_table(cursor, conn):
            cursor.execute("""\
                CREATE TABLE IF NOT EXISTS running_csync (
                csync_pid INTEGER PRIMARY KEY,
                sync_pid INTEGER DEFAULT 0,
                source_path TEXT)""")
        
            conn.commit()
        
        global _DB, _CURSOR, _CONN
        _DB = db_utils.SQLiteConn(_DB_PATH, create_table)
        _CURSOR = _DB.cursor
        _CONN = _DB.conn
        return func(*args, **kwargs)
    return wrapper
```

Reply:
@Michaelvll Slightly updated the logic for this:
```
        if _DB is not None:
            return
```
It was not allowing the database related functions to run. Please confirm if the updated one looks good. I confirmed that it works as intended.